internal/services/oracle/db_system_shapes_data_source.go

```diff
@@ -0,0 +1,219 @@
+// Copyright © 2024, Oracle and/or its affiliates. All rights reserved
+
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2024-06-01/dbsystemshapes"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type DbSystemShapesDataSource struct{}
+
+type DbSystemShapesModel struct {
+	DbSystemShapes []DbSystemShapeModel `tfschema:"db_system_shapes"`
+	Location       string               `tfschema:"location"`
+}
+
+type DbSystemShapeModel struct {
+	AvailableCoreCount                 int64   `tfschema:"available_core_count"`
+	AvailableCoreCountPerNode          int64   `tfschema:"available_core_count_per_node"`
+	AvailableDataStorageInTbs          int64   `tfschema:"available_data_storage_in_tbs"`
+	AvailableDataStoragePerServerInTbs float64 `tfschema:"available_data_storage_per_server_in_tbs"`
+	AvailableDbNodePerNodeInGbs        int64   `tfschema:"available_db_node_per_node_in_gbs"`
+	AvailableDbNodeStorageInGbs        int64   `tfschema:"available_db_node_storage_in_gbs"`
+	AvailableMemoryInGbs               int64   `tfschema:"available_memory_in_gbs"`
+	AvailableMemoryPerNodeInGbs        int64   `tfschema:"available_memory_per_node_in_gbs"`
+	CoreCountIncrement                 int64   `tfschema:"core_count_increment"`
+	MaxStorageCount                    int64   `tfschema:"max_storage_count"`
+	MaximumNodeCount                   int64   `tfschema:"maximum_node_count"`
+	MinCoreCountPerNode                int64   `tfschema:"min_core_count_per_node"`
+	MinDataStorageInTbs                int64   `tfschema:"min_data_storage_in_tbs"`
+	MinDbNodeStoragePerNodeInGbs       int64   `tfschema:"min_db_node_storage_per_node_in_gbs"`
+	MinMemoryPerNodeInGbs              int64   `tfschema:"min_memory_per_node_in_gbs"`
+	MinStorageCount                    int64   `tfschema:"min_storage_count"`
```

Comment:
Some properties here spell min out to minimum and max out to maximum, we prefer to spell these out, so can we be consistent here and spell out min/max to minimum/maximum

Reply:
Done!